pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -820,40 +817,30 @@ async def _handle_file_search_grounding_metadata_streaming(
             )
             self._file_search_tool_call_id = None
 
-    def _handle_executable_code_streaming(
-        self, executable_code: ExecutableCode
-    ) -> tuple[ModelResponsePart | None, str | None]:
+    def _handle_executable_code_streaming(self, executable_code: ExecutableCode) -> ModelResponsePart | None:
         """Handle executable code for streaming responses.
 
-        Returns a tuple of (part_obj, code_execution_tool_call_id).
-        part_obj is None if the part was already generated (to avoid duplicates).
-        code_execution_tool_call_id is None for file_search, or the new ID for code_execution.
+        Returns a part object, or None if the part was already generated (to avoid duplicates).
+        Sets self._code_execution_tool_call_id or self._file_search_tool_call_id as appropriate.
         """
         code = executable_code.code
-        if not code:  # pragma: no cover
-            code_execution_tool_call_id = _utils.generate_tool_call_id()
-            part_obj = _map_executable_code(executable_code, self.provider_name, code_execution_tool_call_id)
-            return part_obj, code_execution_tool_call_id
-
         has_file_search_tool = any(
             isinstance(tool, FileSearchTool) for tool in self.model_request_parameters.builtin_tools
         )
 
-        if has_file_search_tool and (file_search_query := _extract_file_search_query(code)):
+        if code and has_file_search_tool and (file_search_query := _extract_file_search_query(code)):
             if self._file_search_tool_call_id is None:
```

Comment:
Do we need this check? Or could we override if it's already set, like we do below for `self._code_execution_tool_call_id`?